File: cpp/LeetCode/No1299.cpp
/*
Given an array arr
, replace every element in that array with the greatest element among the elements to its right
, and replace the last element with -1.

After doing so, return the array.
*/

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int index = arr.size()-2;
        int temp_max = arr[arr.size()-1];
        arr[arr.size()-1] = -1;
        while(index>-1){
            if(arr[index] > temp_max){
                int temp2 = arr[index];
                arr[index] = temp_max;
                temp_max = temp2;
            }
            else{
                arr[index] = temp_max;
            }
            index--;
        }
        return arr;
    }
};

int main(void){
    vector<int> arr = {17,18,5,4,6,1};
    Solution solution;
    solution.replaceElements(arr);

    for(int num : arr){
        cout << num << ",";
    }
    cout << endl;

    return 0;
}

File: python/LeetCode Quiz/No1299.py
'''
Given an array arr
, replace every element in that array with the greatest element among the elements to its right
, and replace the last element with -1.

After doing so, return the array.
'''
from typing import List

class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        i = len(arr)-2
        temp = arr[-1]
        while i>-1:
            if arr[i]>temp:
                temp2 = arr[i]
                arr[i] = temp
                temp = temp2
            else:
                arr[i] = temp
            i-=1
        arr[-1] = -1
        return arr

arr = [56903,18666,60499,57517,26961]
print(Solution.replaceElements(Solution(), arr))